### PR TITLE
Create a function for adding unittests in CMake.

### PR DESCRIPTION
**Pull request by antiagainst**
Also remove unnecessary main() function for unittests.

### CODE REVIEW

**Comment by antiagainst** [comment]
This is the first commit in #219.

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -24,6 +24,29 @@
...
+    target_compile_options(${target} PRIVATE /wd4503)
+  endif()
+  target_include_directories(${target} PRIVATE
+    ${spirv-tools_SOURCE_DIR}
+    ${gtest_SOURCE_DIR}/include
+    ${gmock_SOURCE_DIR}/include
+  )
+  target_link_libraries(${target} PRIVATE ${libs})
+  target_link_libraries(${target} PRIVATE gmock_main)
+  add_test(NAME spvtools_${target} COMMAND ${target})
```
Comment (dneto0) [inline test/CMakeLists.txt]: The test name is inconsistent with the old style. 
Before we had spirv-tools-... and now its spvtools_
I like the hyphen (I think we used that elsewhere too), and I like spelling out spirv-tools.

**Comment by dneto0** [comment]
-1 just for the naming of the test.

File: test/CMakeLists.txt (hunk 1 above)
Comment (antiagainst) [inline test/CMakeLists.txt]: Done.

**Comment by dneto0** [comment]
+2

**Pull request merged**